### ISSUE TITLE
Docs: [no-await-in-loop] mention preventing unhandled rejections as an additional benefit of the rule

### ISSUE THREAD

**Issue by kirkwaiblinger** [eslint/eslint#18918]
### Docs page(s)

https://eslint.org/docs/latest/rules/no-await-in-loop

### What documentation issue do you want to solve?

The docs currently understate the value of the rule ❤ by not mentioning promise rejections

### What do you think is the correct solution?

Mention promise rejections

### Participation

- [x] I am willing to submit a pull request for this change.

### Additional comments

Specifically, it's usually considered bad to write code like this...
```js
async function asyncFunctionThatMightThrow() {
  const promise1 = Promise.reject(new Error('rejected 1'));
  const promise2 = Promise.reject(new Error('rejected 2'));
 
  // bad
  const value1 = await promise1;
  const value2 = await promise2;

  return [value1, value2];
}
```
...because even if you're aware of exceptions/rejections when calling it...
```js
async function foo() {
    try {
      return await asyncFunctionThatMightThrow();
    } catch (e) {
        // gotcha!
       return "don't worry, situation is under control!";
    }
}
```
...you still are punished with floating unhandled rejections:
```js

foo(); // returns "don't worry, situation is under control!";

// then, node process crashes, Uncaught Error: rejected 2
```

But, with the promise concurrency methods, all is well
```js
async function asyncFunctionThatMightThrow() {
  const promise1 = Promise.reject(new Error('rejected 1'));
  const promise2 = Promise.reject(new Error('rejected 2'));
 
  // good
  const [value1, value2] = await Promise.all([promise1, promise2])

  return [value1, value2];
}

// same foo
foo(); // returns "don't worry, situation is under control!"

// no crash!
```

Now, if I rewrite this in a way that handles arbitrarily many promises...
```js
async awaitInLoop(n) {
    const arrayOfRejections = new Array(n).fill(undefined).map((_, i) => Promise.reject(new Error("rejected " + i)));
    const returnArray = [];
    for (const promise of arrayOfRejections) {
          const promisedValue = await promise; // flagged by no-await-in-loop 😄 
          returnArray.push(promisedValue);
    }
    return returnArray;
}
```
...and repeat the exercise from before, I'll get floating unhandled rejections. Instead, using a promise concurrency method, ie the following....
```js
async awaitInLoop(n) {
    const arrayOfRejections = new Array(n).fill(undefined).map((_, i) => Promise.reject(new Error("rejected " + i)));
    const returnArray = [];
    for (const promisedValue of await Promise.all(arrayOfRejections)) {
          returnArray.push(promisedValue);
    }
    return returnArray;
}
```
...we are safe from floating unhandled rejections again.

***

I think this is an additional benefit to the rule worthy of mention.

**Comment by nzakas** [comment]
Sounds like a good improvement to the docs to me. Feel free to open a PR.
